### DISCUSSION TITLE
[Stream Deck Plus] How to control volume of a single application with dials?

### DISCUSSION THREAD

**Discussion by koeberlue** [StreamController/StreamController#157] [Q&A]
I would like to use the dials of the Stream Deck Plus to control the volume of single applications. So my preferred setup would be:

1. Dial: Control master volume
2. Dial: Control Spotify volume
3. Dial: Control Zoom volume
4. Dial: Control Firefox volume

Is there already a simple way to do this? Would I need to write a custom plugin? Or is there a way to call a custom script (like [this](https://askubuntu.com/a/1418749) ) depending on the direction of the dial rotation?

**Comment by Core447** [comment]
Unfortunately there's currently no easy / plugin-and-play solution for this. However, you can call different scripts per dial rotation. You need to add 2 (or 3 if you also want a mute script) RunCommand actions to each dial and use the EventAssigner in the action config area (where you can specify the command) to configure when the action should trigger. This is not ideal, but it's a working workaround for now.

**Comment by batistein** [comment]
@koeberlue have you solved this use-case with the mentioned script approach? It would be great if you don't mind sharing your solution.

**Reply by koeberlue** [reply]
Sorry, I had to put this on my backlog for now. I am currently using the VolumeMixer to control my headphones and use the default OS tooling to control individual apps with my mouse.

**Comment by Core447** [comment]
An alternative might be to use the VolumeMixer plugin, which now comes with dial support

**Comment by Cfomodz** [comment]
Install the **VolumeMixer** plugin from the StreamController plugin store. It now includes native dial support for per-application volume control.

### Setup

1. Open StreamController → Plugin Store
2. Install "VolumeMixer"
3. Drag the "Dial" action to each Stream Deck Plus dial
4. Each dial automatically controls a different active audio application

### Usage

- **Rotate:** Adjust volume
- **Press:** Toggle mute
- **Navigate:** Use MoveLeft/MoveRight button actions to shift between applications

### Limitation

Applications are auto-detected from PulseAudio in dynamic order. You cannot assign specific apps (e.g., "Spotify always on Dial 2").

---

## Alternative: Fixed Application Assignment

For dedicated per-app control, use **RunCommand actions** with custom scripts:

1. Add 2 RunCommand actions per dial (volume up/down)
2. Configure each with a script targeting your specific app via `pactl`
3. Use **EventAssigner** to bind:
   - Clockwise rotation → volume up script
   - Counter-clockwise → volume down script

Example script snippet:
```bash
pactl set-sink-input-volume $(pactl list sink-inputs | grep -B 20 "Spotify" | grep "Sink Input" | cut -d'#' -f2) +5%
```

This requires scripting but gives you fixed dial assignments.
